packages/paste-core/components/badge/src/index.tsx

```diff
@@ -1,49 +1,60 @@
 import * as React from 'react';
 import * as PropTypes from 'prop-types';
-import {Box} from '@twilio-paste/box';
+import {Box, safelySpreadBoxProps} from '@twilio-paste/box';
+import {ErrorIcon} from '@twilio-paste/icons/esm/ErrorIcon';
 
 import type {BadgeProps, BadgeVariants} from './types';
 import {useResizeChildIcons} from './hooks';
-import {badgeFocusableStyles, badgeVariantStyles} from './styles';
-import {getBadgeSpanProps, isFocusableElement} from './utils';
-import {BadgeWrapper} from './BadgeWrapper';
+import {badgeVariantStyles, getBadgeAnchorStyles, getBadgeButtonStyles} from './styles';
 
 export type {BadgeProps};
 
 export const Badge = React.forwardRef<HTMLElement, BadgeProps>(
-  ({variant, children, element = 'BADGE', ...props}, ref) => {
+  ({as, variant, children, element = 'BADGE', ...props}, ref) => {
     const resizedChildren = useResizeChildIcons(children);
 
-    const variantStyles = badgeVariantStyles[variant];
-    const spanProps = getBadgeSpanProps(props);
-    const isFocusable = isFocusableElement(props);
+    let badgeStyles = badgeVariantStyles[variant];
+
+    if (as != null) {
+      if (as === 'a') {
+        badgeStyles = {...badgeStyles, ...getBadgeAnchorStyles()};
+      }
+      if (as === 'button') {
+        badgeStyles = {...badgeStyles, ...getBadgeButtonStyles(variant)};
+      }
+    }
 
     return (
-      // @ts-expect-error we need to explore polymorphic types for this ref to work https://www.benmvp.com/blog/forwarding-refs-polymorphic-react-component-typescript/
-      <BadgeWrapper {...props} ref={isFocusable ? ref : null}>
-        <Box
-          {...spanProps}
-          alignItems="center"
-          as="span"
-          border="unset"
-          borderRadius="borderRadius30"
-          columnGap="space10"
-          display="flex"
-          element={element}
-          fontSize="fontSize20"
-          fontWeight="fontWeightSemibold"
-          lineHeight="lineHeight10"
-          maxWidth="max-content"
-          paddingX="space30"
-          paddingY="space20"
-          variant={variant}
-          ref={!isFocusable ? ref : null}
-          {...variantStyles}
-          {...(isFocusable && {...badgeFocusableStyles})}
-        >
-          {resizedChildren}
-        </Box>
-      </BadgeWrapper>
+      <Box
+        {...safelySpreadBoxProps(props)}
```

Comment:
suggestion(blocking): One thing I forgot, even though we aren't doing the external link icon, we should still be doing the secureExternalLink functionality from anchor still. 

You can see this in display pill https://github.com/twilio-labs/paste/blob/0cd424df4d3ed57e60a0d93b97dda8b4c708ce7c/packages/paste-core/components/display-pill-group/src/DisplayPill.tsx#L31

Where if there is an href, we use the imported anchor utility to add a little security to the link.

Reply:
ah great catch, thank you!